core/src/tests/test_config_parser_dir.cc

```diff
@@ -29,8 +29,79 @@
 #include "lib/parse_conf.h"
 #include "dird/dird_globals.h"
 #include "dird/dird_conf.h"
+#include "lib/output_formatter_resource.h"
 
 namespace directordaemon {
+bool DoReloadConfig() { return false; }
+}  // namespace directordaemon
+
+namespace directordaemon {
+
+static std::string sprintoutput{};
+
+bool sprintit(void* ctx, const char* fmt, ...)
+{
+  va_list arg_ptr;
+  PoolMem msg;
+
+  va_start(arg_ptr, fmt);
+  msg.Bvsprintf(fmt, arg_ptr);
+  va_end(arg_ptr);
+
+  sprintoutput += msg.c_str();
+  return true;
+}
+
+TEST(ConfigParser_Dir, ParseSchedulerOddEvenDaysCorrectly)
+{
+  OSDependentInit();
+
+  std::string path_to_config_file = std::string(
+      RELATIVE_PROJECT_SOURCE_DIR "/configs/bareos-configparser-tests");
+  my_config = InitDirConfig(path_to_config_file.c_str(), M_ERROR_TERM);
+  my_config->ParseConfig();
+
+  ScheduleResource *resource_iterator = nullptr, *even_schedule = nullptr,
+                   *odd_schedule = nullptr;
+
+  do {
+    std::string str;
+    resource_iterator
+        = (directordaemon::ScheduleResource*)my_config->GetNextRes(
+            R_SCHEDULE, resource_iterator);
+    str += resource_iterator->resource_name_;
+    if (str == "Even Weeks") {
+      even_schedule = resource_iterator;
+    } else if (str == "Odd Weeks") {
+      odd_schedule = resource_iterator;
+    }
+  } while (resource_iterator->next_ != nullptr);
```

Comment:
This whole loop probably can be replaced by two my_config->GetResWithName() calls.